Associative-Containers/exercise2.cpp

```diff
@@ -1,85 +1,141 @@
-#include<map>
-#include<iostream>
-#include<numeric>
-#include<cmath>
-#include<functional>
-#include<cassert>
+#include <map>
+#include <iostream>
+#include <numeric>
+#include <cmath>
+#include <functional>
+#include <cassert>
+#include <iomanip>
+enum class ErrorCode
+{
 
-enum class ErrorCode{
-    
     Ok,
     BadCharacter,
     BadFormat,
     DivideBy0,
     SqrtOfNegativeNumber,
     ModuleOfNonIntegerValue,
-    None 
-    };
+    None
+};
+bool DivideBy0 (const std::string &input_)
+{
+    for(size_t i = 0;i<input_.size();i++)
+        {
+            // std::cout<<input_[i];
+            if (input_[i] == '/' && input_[i+1] == '0'){return false;}
+        }   
+        return true;
+}
+ErrorCode process(std::string input, double *out)
+{
 
-ErrorCode process(std::string input, double* out){ 
-    
-    std::map<char, std::function<double(double,double)>> core ={
-        {'+',[](double a, double b){return a+b;}},
-        {'-',[](double a, double b){return a-b;}},
-        {'/',[](double a, double b){return a/b;}},
-        {'*',[](double a, double b){return a*b;}},
-        {'%',[](double a, double b){return static_cast<int>(a) % static_cast<int>(b);}},
+    std::map<char, std::function<double(double, double)>> core = {
+        {'+', [](double a, double b)
+         { return a + b; }},
+        {'-', [](double a, double b)
+         { return a - b; }},
+        {'/', [](double a, double b)
+         { return a / b; }},
+        {'*', [](double a, double b)
+         { return a * b; }},
+        {'%', [](double a, double b)
+         { return static_cast<int>(a) % static_cast<int>(b); }},
         // {"!",[](int a, int b){return ;}},
-        {'^',[](double a, double b){return std::pow(a,b);}}
-    };
-
-    // std::cout << core[op](x,y)<<'\n';
+        {'^', [](double a, double b)
+         { return std::pow(a, b); }}};
+    std::stringstream streaminput(input);
```

Comment:
```suggestion
         { return std::pow(a, b); }}};
    std::stringstream input_stream(input);
```
As in the English language the modifier (In Polish przydawka) **must be before the noun, which it refers to**

Reply:
Uploaded